byzcoin/contracts/coins.go

```diff
@@ -114,6 +123,10 @@ func (c *contractCoin) Invoke(rst byzcoin.ReadOnlyStateTrie, inst byzcoin.Instru
 			cid string
 			did darc.ID
 		)
+		if inst.InstanceID.Equal(byzcoin.NewInstanceID(target)) {
+			err = errors.New("cannot send coins to ourselves")
```

Comment:
Because my brother-in-law made fun of me when his coins disappeared that he sent to himself ;) But in fact a valid question: shall we throw an error here or just ignore it? 

I suppose it's most probably an error if you send coins to yourself, so it's nice to be alerted.

On the other hand you could prove that you have this many coins - but then again, you can simply look them up, we're not using any zero-proofs.

So I propose to keep the error, because it might help somebody debug something sometimes in the future. 

But I'm open to removing it and only throw the error if we want to send more coins to ourselves than the account has, and ignore it otherwise.

Reply:
Ok, just wanted to know your thoughts, leave it like this, I think.